### ISSUE TITLE
Inheritance may cause false positives

### ISSUE THREAD

**Issue by steffsas** [CROSSINGTUD/CryptoAnalysis#327]
**Prerequisites**
Currently, I'm testing the CryptoAnalysis on non-crypto APIs. Therefore, I wrote a custom API and a respective CrySL rule for testing purposes only. Custom CrySL rules are recognized, also on self-written APIs.

**Bug**
At the moment I am facing a problem with inheritance. It seems that methods that are not explicitly declared in a class are not recognized by CogniCrypt for further analysis. In the example below, a misuse is indicated in the line where __c.supermethod()__ is called: " InheritanceAPI.Subclass object not completed. Expected call to InheritanceAPI.Superclass: void supermethod()". However, this is clearly not a misuse. If the method __supermethod()__ is declared explicitly in class __Sublcass.java__, everything works fine.

**_main.java_**
```
public class Main {
	public static void main(String[] args) {
		Subclass c = new Subclass();
		c.method();
		c.supermethod();
	}
}	
```

**_Superclass.java_**
```
public class Superclass {
	public void supermethod() {};
}
```

**_Subclass.java_**
```
public class Subclass extends Superclass {
	public void method() {};
}
```

and the following CrySL rule:

**_Subclass.crysl_**
```
SPEC InheritanceAPI.Subclass

OBJECTS
	java.lang.String test;
	
EVENTS
	m: method();
	s: supermethod();
	
ORDER
	m, s
```
